In [1]:
# AIzaSyBIa0cVjNajRLivM814t84sHxETeawxvf0  # get it from google ai gemini.

In [2]:
!pip install PyPDF2

   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 232.6/232.6 kB 2.0 MB/s eta 0:00:00


In [3]:
import google.generativeai as genai
import PyPDF2 as pdf
import json

In [4]:
# Call Gemini model.
genai.configure(api_key = "AIzaSyBIa0cVjNajRLivM814t84sHxETeawxvf0")

In [5]:
# Few functions to deal with PDF and connect to Gemini api key.
# Get some response from text using the given functions.

def get_gemini_repsonse(input):
    model=genai.GenerativeModel('gemini-pro')
    response=model.generate_content(input)
    return response.text

def input_pdf_text(uploaded_file):
    reader=pdf.PdfReader(uploaded_file)
    text=""
    for page in range(len(reader.pages)):
        page=reader.pages[page]
        text+=str(page.extract_text())
    return text

In [6]:
get_gemini_repsonse("Hi")

# Model is working fine.

'Hello! How can I help you?'

In [7]:
def build_prompt(text):
    input_prompt=f"""
    As the Chief Financial Officer (CFO) of a company,
    your role is to present a summary of the company's financial
    results for the following details.
    Provide an overview that includes key metrics such as revenue,
    net income, operating expenses, and cash flow.
    Highlight significant achievements or setbacks,
    compare the results with the previous period,
    and identify any trends or factors that influenced the
    performance. Your summary should be clear, data-driven,
    and provide insights for stakeholders to understand the
    company's financial health and strategic direction
    Results Document:{text}

    Provide the summary as a list.
    I want the response in one single string having the structure
    {{"Summary:[]"}}
    """
    return input_prompt

In [8]:
text=input_pdf_text("/content/HDFC_Results.pdf")

In [9]:
input_prompt = build_prompt(text)

In [10]:
response=get_gemini_repsonse(input_prompt)
response = json.loads(response)

In [11]:
import textwrap


def add_newlines(text):
  sentences = text.split('. ')
  return '.\n'.join(sentences) + '.\n'

def print_response(response):
  print("Summary:")
  for suggestion in response["Summary"]:
      print("- {}".format(add_newlines(suggestion)))

print_response(response)

Summary:
- HDFC Bank reported a 26.4% increase in net interest income in Q1 FY 2024-25, rising from ₹ 236.0 bn to ₹ 298.4 bn compared to Q1 2023-24.
The balance sheet size expanded by 42.6% to ₹ 35,673 bn from ₹ 25,017 bn during the same period.
Total advances witnessed a 52.5% growth to reach ₹ 23,791 bn as of Q1 FY 2024-25, while total deposits saw a 24.4% rise to ₹ 161.7 bn.
The bank's net profit surged by 35.3% to ₹ 70.5 bn in Q1 FY 2024-25.
Fees and commission income increased by 12.0% to ₹ 62.9 bn.
The gross NPA ratio stood at 1.33%, whereas the CASA (Current Account Savings Account) ratio was 36%.
The bank had 8,851 branches, with plans to open 515 new branches by the end of March 2025.
The total number of banking correspondents stood at 15,146.
Retail advances accounted for 50% of the total advances, with 31% being contributed by CRB Corporate and 19% by Net Interest Margin.
The bank's capital adequacy ratio was 19.3%.
HDFC Bank recorded a strong performance in the first quarte

Summary:
- **Overview:** HDFC Bank Ltd.
presented strong financial performance in Q1 FY 2024-25 with key metrics showing positive growth.

**Key Metrics:**
* Net Interest Income increased by 26.4% to ₹298.4 billion from ₹236.0 billion in Q1 of the previous year.
* Total advances grew by 24.4% to ₹23,791 billion from ₹19,131 billion.
* Total deposits increased by 35.3% to ₹25,017 billion from ₹18,507 billion.
* Net profit rose by 52.5% to ₹119.5 billion from ₹78.3 billion.

**Significant Achievements:**
* Net interest margin improved to 3.7%, contributing to higher net interest income.
* Retail advances grew by 100.4%, driven by strong demand for mortgages and personal loans.
* CASA ratio increased to 36%, reflecting a shift towards lower-cost deposits.
* Fee and commission income grew by 12.0% to ₹62.9 billion.

**Challenges:**
* Gross NPA ratio increased slightly to 1.33% from 1.23% in the previous quarter, reflecting the impact of the economic slowdown.

**Outlook:** HDFC Bank remains optimistic about the future, focusing on increasing its market share in retail and corporate banking while managing risks prudently..